.github/workflows/OCV-Git-Cache.yaml

```diff
@@ -0,0 +1,41 @@
+name: OCV Git cache
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/OCV-Git-Cache.yaml'
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  Update:
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [opencv-cn-lin-x86-64, opencv-cn-lin-arm64, opencv-cn-win, opencv-cn-mac-arm64, opencv-cn-mac-x86-64]
+    steps:
+      - name: Setup GIT_CACHE_PATH Environment variable
+        run: |
+          if [[ "${{ matrix.os }}" == *"opencv-cn-lin"* ]]
+          then
+            echo "GIT_CACHE_PATH=/home/opencv-cn/git_cache" >> $GITHUB_ENV
```

Comment:
Can we define these paths on the runner side? E.g. setup env before runner launch.

Reply:
Defined variables on host side for Linux and Windows, MacOS left. Need to investigate how to do it properly. I've already tried several ideas, but without solution at the moment ):

Reply:
> but without solution at the moment

Same. I tried several methods but none of that worked.

Reply:
So I can confirm that we can set environment variables using `env.sh` from GitHub Actions scripts. You can see my example here: https://github.com/fengyuentau/opencv/pull/8/checks.

Basically what you need to do is:
1. Have environment variables exported. For example, `export OPENCV_CI_HOME=/Users/opencv-cn/Workspace/opencv-ci`. I have them set already in `.zshrc`, which you can edit as well.
1. Add custom environment variables into `varCheckList` in `env.sh` in each folder of GitHub Actions scripts.
1. Re-run `config.sh`. The tricky thing is I did not find an option in `config.sh` to reload environment, so I need your help to remove and re-run `config.sh`.

Reply:
That's great! Thank you @fengyuentau for your experiments and solution!

I want to confirm only one moment, did you run a runner as a service (`svc.sh` script) or using `run.sh` script?

Reply:
By the way, I re-registered runners with updated `env.sh` script. I'll check variables in workflows.

Reply:
@fengyuentau, I've tested it with workflows in this PR and variables are defined properly for runners as services.

Thank you so much!

Reply:
Great! `env.sh` is called in the last line of `config.sh`, so it does not matter whether a runner is run as a service or using `run.sh` script. Let me adjust my workflows for macOS tomorrow.

Reply:
Yes, I've seen it. I'd tried few things with environment variables and runners which had worked when I'd called `run.sh` script, but had not when I'd called `svc.sh`.

But now it works with `env.sh` and `.zshrc` what is great.

Reply:
Did you make it work with calling `run.sh` or `svc.sh`? I did not expect this since `env.sh` is sourced in `config.sh`.

Reply:
I made it work calling `run.sh` script. As I understand, it worked because when you call `run.sh` script, it uses an environment for current user, but if to call `svc.sh`, it uses an environment for services.

Reply:
Thanks for the information. I filed an issue to github actions and hopefully can get an answer there.

Reply:
Thank you!